javascript/javascript3/week2/index.js

```diff
@@ -0,0 +1,79 @@
+console.log("js3 week2");
+//Movies exercise
+//bad movies
+const badMoviesPromise = fetch("https://gist.githubusercontent.com/pankaj28843/08f397fcea7c760a99206bcb0ae8d0a4/raw/02d8bc9ec9a73e463b13c44df77a87255def5ab9/movies.json")
+    .then(response => response.json())
+    .then(movieResponse=> movieResponse.filter(movie =>movie.rating <= 4));
+    
+    badMoviesPromise
+        .then(console.log);
+
+//bad movies sinec 2000
+const badMoviesPromiseSinceXyear = fetch("https://gist.githubusercontent.com/pankaj28843/08f397fcea7c760a99206bcb0ae8d0a4/raw/02d8bc9ec9a73e463b13c44df77a87255def5ab9/movies.json")
+    .then(response => response.json())
+    .then(movieResponse=> movieResponse.filter(movie =>movie.year >= 2000 ))
+    .then(lastResponse => lastResponse.filter(movie=>movie.rating<= 4))
+    
+    badMoviesPromiseSinceXyear
+        .then(console.log);
+
+//Promise that resolves after set time exercise
+function resolveAfter (resolveAfter) {
+    return new Promise ((resolve, reject) => {
+        setTimeout(() => {
+            resolve(`logged after ${resolveAfter} second (s)`)
+        }, resolveAfter);
+    })
+
+}
+resolveAfter(2000).then((mess)=> {
+    console.log(mess);
+})
+
+//Promise that resolves after set time exercise with async/awaity sytx
+
+async function resolveAfterAsync (resolveAfter) {
+    return await new Promise ((resolve, reject) => {
+        setTimeout(() => {
+            resolve(`logged after ${resolveAfter} second(s) using asyc/await`)
+        }, resolveAfter);
+    })
+
+}
+resolveAfterAsync(2000).then((mess)=> {
+    console.log(mess);
+})
+
+//Fetching and waiting exersice
+function getData (time)  {
+    return new Promise((resolve, reject)=> {
+        setTimeout(() => {
+            const  result = fetch("https://dog.ceo/api/breeds/image/random")
+            .then(response => response.json())
+            resolve(result)
+        }, time);
+    })
+}
+getData(5000).then((data)=> {
+    console.log(data)
+    
+})
+//Fetching and waiting exercise usung async/await
+async function getDataAsync (time)  {
+    try {
+        return await new Promise((resolve, reject)=> {
+            setTimeout(() => {
+                const  result = fetch("https://dog.ceo/api/breeds/image/random")
+                .then(response => response.json())
+                resolve(result)
+                console.log(result)
+            }, time); 
+        })
```

Comment:
What's being printed to the console here isn't the actual data but the promise instance. The reason why `result` doesn't contain the data is because we didn't put `await` keyword before `fetch`. But if we do that we'll get an error saying `await can only be used in async function`. The reason for that is that the callback in `setTimeout` is not an `async` function.
So my suggestion is this: we don't need to use a callback here thanks to the power of `await`. We can wait for our `resolveAfterAsync` method to finish using `await` then call the `fetch` with `await` as well to get the data.